### PR TITLE
PRO-2031 Archanova Error

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-2039/story-task-archanova-error

## Description

- Fixed: Multiple times Archanova init
- Fixed: Archanova disconnect error

## How Has This Been Tested?

- Tested on Android device

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/services/archanova.js [hunk 1]
```diff
@@ -170,10 +170,12 @@ const parseEstimateError = (error: Error, logExtras: Object = {}) => {
 export class ArchanovaService {
   sdk: Sdk = null;
   sdkInitialized: boolean;
+  privateKey: string;
 
   constructor() {
     this.sdk = null;
     this.sdkInitialized = false;
+    this.privateKey === '';
```
Comment (IAmKio) [inline src/services/archanova.js:178]: Is this supposed to be `=== ''`?

**Review by IAmKio — CHANGES_REQUESTED** [review]
Am i right in thinking that the only thing that is changed here is just the private key initialisation?

File: src/services/archanova.js (hunk 1 above)
Comment (lalitkumarrock) [reply src/services/archanova.js:178]: My bad, I'll fix it ASAP.

**Comment by lalitkumarrock** [comment]
> Am i right in thinking that the only thing that is changed here is just the private key initialisation?

This issue occurs when we have already initialized the sdk and try to restart again.
The private key compares the previous key and the next key for initialization.

**Review by IAmKio — APPROVED** [review]
LGTM

**Pull request merged**